Core/Executor/UserManager.php

```diff
@@ -111,25 +115,29 @@ protected function update()
 
                 $assignedGroups = $userService->loadUserGroupsOfUser($user);
 
+                $targetGroupIds = [];
                 // Assigning new groups to the user
                 foreach ($this->dsl['groups'] as $groupToAssignId) {
+                    $groupId = $this->referenceResolver->resolveReference($groupToAssignId);
+                    $groupToAssign = $this->userGroupMatcher->matchOneByKey($groupId);
```

Comment:
This might lead to a slightly  slower performance as before not every group was loaded from storage, but otherwise we can not cover all required checks below...